pkg/ansible/proxy/proxy.go

```diff
@@ -238,15 +246,31 @@ func addWatchToController(owner kubeconfig.NamespacedOwnerReference, cMap *contr
 			awMap.Store(resource.GroupVersionKind())
 			typeString := fmt.Sprintf("%v.%v", owner.Kind, ownerGV.Group)
 			log.Info("Watching child resource", "kind", resource.GroupVersionKind(), "enqueue_annotation_type", typeString)
-			err = contents.Controller.Watch(&source.Kind{Type: resource}, &osdkHandler.EnqueueRequestForAnnotation{Type: typeString})
+			err = addWatch(contents.Controller, &source.Kind{Type: resource}, &osdkHandler.EnqueueRequestForAnnotation{Type: typeString})
 			if err != nil {
+				log.Error(err, "GVK", resource.GroupVersionKind())
 				return err
 			}
 		}
 	}
 	return nil
 }
 
+func addWatch(c controller.Controller, s source.Source, eh handler.EventHandler, predicates ...predicate.Predicate) error {
+	errChan := make(chan error, 1)
+	go func() {
+		err := c.Watch(s, eh, predicates...)
```

Comment:
We should work upstream to get a `controller.WatchWithContext()` function so that we don't need to resort to this.

The problem here is that it is not possible to cancel the `c.Watch()` call, so if we reach the deadline, the `c.Watch` call will just stay running in the background forever. This is effectively a resource leak.